### DISCUSSION TITLE
About `--tolerate-republish`

### DISCUSSION THREAD

**Discussion by alecmev** [yarnpkg/berry#4770] [Q&A] [answered]
### Self-service

- [x] I'd be willing to implement a fix, if this needs fixing

### Describe the bug

The word "tolerate" [means](https://www.google.com/search?q=define+tolerate):

> Allow the existence, occurrence, or practice of (something that one dislikes or disagrees with) without interference.

So "tolerate republish" would mean "it's okay to republish". But the description of the flag suggests otherwise:

https://github.com/yarnpkg/berry/blob/ed919239529c71e2cc4419d970b0d57dde1a6915/packages/plugin-npm-cli/sources/commands/npm/publish.ts#L37-L39

And so does the implementation:

https://github.com/yarnpkg/berry/blob/ed919239529c71e2cc4419d970b0d57dde1a6915/packages/plugin-npm-cli/sources/commands/npm/publish.ts#L70-L91

However, this is out for [over three years](https://github.com/yarnpkg/berry/pull/199), so I feel like this would've been caught by now. Am I missing something?

**Comment by arcanis** [comment, marked as answer]
Publishing a package over an existing version isn't possible on registries - they would crash, and the process would exit with an error code. This is problematic when using `yarn workspaces foreach npm publish`, as it could stop the iteration.

Adding `--tolerate-republish` tells Yarn that if the package already exists, then it's fine to skip the "republish" error and exit without error code. To this end we check for the package existence first, since it allows us to skip the `yarn pack` call for all packages that don't need publishing.

**Reply by alecmev** [reply]
Thank you! Sorry for not posting this in the discussions in the first place. Would it make sense to rename this flag to `--tolerate-conflict-error` perhaps, or something like that? I'm new to publishing, so it didn't even occur to me that the registries are append-only. Or at least a mention in the docs, saying that a `PUT` of an existing version always results in a 409, and the use case you described.

**Reply by arcanis** [reply]
I don't think the value is high enough to be worth a breaking change, tbh 😅

**Reply by alecmev** [reply]
That's true! Is a doc update welcome though?
